x/ecocredit/client/testsuite/grpc.go

```diff
@@ -387,42 +301,44 @@ func (s *IntegrationTestSuite) TestGetSellOrder() {
 			resp, err := rest.GetRequest(tc.url)
 			require.NoError(err)
 
-			var sellOrder ecocredit.QuerySellOrderResponse
-			err = val.ClientCtx.Codec.UnmarshalJSON(resp, &sellOrder)
-
-			if tc.expErr {
-				require.Error(err)
-				require.Contains(string(resp), tc.errMsg)
-			} else {
-				require.NoError(err)
-				require.NotNil(sellOrder.SellOrder)
-			}
+			var res marketplace.QuerySellOrderResponse
+			err = val.ClientCtx.Codec.UnmarshalJSON(resp, &res)
+			require.NoError(err)
+			require.NotNil(res.SellOrder)
+			require.Equal(res.SellOrder.Quantity, "10")
 		})
 	}
 }
 
 func (s *IntegrationTestSuite) TestGetSellOrders() {
 	val := s.network.Validators[0]
+	_, _, batchDenom := s.createClassProjectBatch(val.ClientCtx, val.Address.String())
+	validAsk := types.NewInt64Coin(core.DefaultParams().AllowedAskDenoms[0].Denom, 10)
+	expiration, err := types2.ParseDate("expiration", "2090-10-10")
+	s.Require().NoError(err)
+	_, err = s.createSellOrder(val.ClientCtx, &marketplace.MsgSell{
+		Owner: val.Address.String(),
+		Orders: []*marketplace.MsgSell_Order{
+			{BatchDenom: batchDenom, Quantity: "10", AskPrice: &validAsk, Expiration: &expiration},
+			{BatchDenom: batchDenom, Quantity: "10", AskPrice: &validAsk, Expiration: &expiration},
+			{BatchDenom: batchDenom, Quantity: "10", AskPrice: &validAsk, Expiration: &expiration},
+		},
+	})
+	s.Require().NoError(err)
 
 	testCases := []struct {
 		name     string
 		url      string
-		expErr   bool
-		errMsg   string
 		expItems int
 	}{
 		{
 			"valid request",
-			fmt.Sprintf("%s/regen/ecocredit/v1alpha1/sell-orders", val.APIAddress),
-			false,
-			"",
-			3,
+			fmt.Sprintf("%s%ssell-orders", val.APIAddress, marketplaceRoute),
+			-1,
```

Comment:
Expected number of items `-1` ?

Reply:
i used -1 to indicate an unknown amount of items. this test checks the paginated limit, so we can expect 2, however, with no pagination, theres no way of knowing how many sell orders there are without coupling the tests again i.e. tracking when other tests create sell orders.. ? i have seen the pattern of using -1 to indicate infinite or unknown before, but if this isn't so clear, im open to suggestions

Reply:
I think migrating away from integration tests for client tests is where my heads at but maybe we can leave this as is for now and address as a followup with #1041.